[RFR] Feature/77 notes comments

enginehouse opened the pull request:
This is is doubtless going to be handed back as it is mid design - also the design for file upload needs to be reigned in to align with the Gov Uk guidelines

Commit:
Adding routes, containers and supporting modules

Commit:
Adding client-side javascript files - alas, not client React : (

Commit:
Merge branch 'fixes/112-overview-new-app-status' into feature/77-notes-comments

Commit:
Merge branch 'fixes/112-overview-new-app-status' into feature/77-notes-comments

Commit:
Adding jquery and mustache for templating comment section

It needed to go in at some time, might as well get it over with

Commit:
Adding file uploader and commenting javascript

Commit:
Adding classes to file upload and textarea to force full width

Commit:
Comment module and updated javascript

Commit:
Addng a big green complete button to round off proceedings

Commit:
A spot of linting

Commit:
Update to user icons and backgrounds

Commit:
Adding a date input component - feel it is an entity all of its own

Commit:
Merge branch 'develop' into feature/77-notes-comments

# Conflicts:
#	app/shared/components/Icon/style.scss

Commit:
Removing merge residue : )

Commit:
List change - shift the left padding to a .spaced class

Commit:
Problem solving

Commit:
Merge branch 'fixes/112-overview-new-app-status' into feature/77-notes-comments

Commit:
De-breaking routes by mapping props to correct components

Commit:
Adding a shim flag to drop a text label into the file upload component

Commit:
Adding the comment / clear function

Commit:
Removing console.log

Commit:
Calculate total comments

Commit:
Adding a date selector component

Commit:
Adding data attribute to hang a javascript off

Commit:
Merge branch 'develop' into feature/77-notes-comments

enginehouse commented:
@Koded - is adding a deadline to the actual comment part of this sprint? I have it as an option but I haven't made it carry it through to the comment and format it (jQuery is nasty for templating : ))
If not then please can I have a review on this? I need to set webpack up to bundle javascripts and allow me to require the plugins and make proper controllers but I haven't done that yet

Koded commented:
Hi @enginehouse - the deadline functionality is part of story 90, so no, you don't need to complete it as part of 77. 

So i'll review...

Koded commented on app/shared/components/FileUpload/component.jsx:
```diff
@@ -0,0 +1,17 @@
+      <label className='form-label' htmlFor={'file-' + props.id}>
+        {props.label}
+      </label>
+      <input className='form-control' type='file' id={'file-' + props.id} data-file-input/>
```
Inline with the guidance (https://govuk-elements.herokuapp.com/form-elements/#form-file-upload) we really need to use native upload controls.

Koded replied:
(know you made a note of this at the top of the PR)

enginehouse replied:
Agreed - it still requires the shim to return the value of the file input but the control itself is the native one now.

Koded commented on app/server/api/static/application-overview-frequency-comments.yml:
```diff
@@ -0,0 +1,209 @@
+  text: "<p>Sed ut perspiciatis unde omnis iste natus error sit voluptatem accusantium doloremque laudantium, totam rem aperiam, eaque ipsa quae ab illo inventore veritatis et quasi architecto beatae vitae dicta sunt explicabo. Nemo enim ipsam voluptatem quia voluptas sit aspernatur aut odit aut fugit, sed quia consequuntur magni dolores eos qui ratione voluptatem sequi nesciunt. Neque porro quis ...
+  commentBlock:
+    limit: 3
+    deadline:
```
I know they are both being worked on, but to make sure we only QA 77, and that we can complete 77, comment this out until we are ready to complete story 90. 

(also the other block)

enginehouse replied:
Removed the deadline component of this

Commit:
Comment adding - adding focus when opening a comment

Koded commented on app/shared/components/Scripts/component.jsx:
```diff
@@ -0,0 +1,15 @@
+
+const Scripts = () => {
+  return (
+    <div>
```
React 16 allows for rendering without wrapping div, something like:

```
const ScriptItems = () => ([
  <script src='/ui/js/vendor/jquery.3.3.1.min.js'/>
  <script src='/ui/js/plugins/file-upload-shim.js'/>
]);
```

enginehouse replied:
Ahh fabulous - I didn't even realise we had upgraded it : ) - I'll chop that extraneous div out (there may be other instances of this)

enginehouse replied:
Addressed this - removed now

Commit:
Commenting out extra functionality

It exceeds the requirements for user story 77 and the file upload is being masked to make it fit the design; reverting this behaviour but keeping some for mocking the upload

enginehouse commented:
Hello @Koded - have tweaked this request with your updates.
